powerdns/views.py

```diff
@@ -92,6 +102,94 @@ def get_queryset(self):
         return queryset
 
 
+
+class GUIRecordViewSet(OwnerViewSet):
+
+    queryset = Record.objects.all().select_related('owner', 'domain')
+    serializer_class = RecordSerializer
+    filter_fields = ('name', 'content', 'domain')
+    search_fields = filter_fields
+
+    def get_serializer_class(self):
+        if self.request.method in ['POST', 'PUT', 'PATCH']:
+            serializer_class = RecordRequestSerializer
+        else:
+            serializer_class = RecordSerializer
+        print(serializer_class)
+        return serializer_class
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        # TODO:: this code needs some love, any idea?
+        auto_accept = (
+            self.request.user.is_superuser or
+            serializer.instance.domain.unrestricted == True or
+            self.request.user == serializer.instance.domain.owner or
+            request.user.id in serializer.instance.domain.authorisations.values_list('authorised', flat=True) # noqa
+        )
+        if auto_accept:
+            record = serializer.instance.accept()
+            serializer.instance.record = record
+            serializer.instance.save()
+            serializer.data['record'] = record
+            code = status.HTTP_201_CREATED
+            headers = {'Location': serializer.data['record']}
+        else:
+            code = status.HTTP_202_ACCEPTED
+            headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=code, headers=headers)
+
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        instance = get_object_or_404(
+            RecordRequest, record__pk=self.get_object().pk)
```

Comment:
are you sure it'd work? as far as i understand it, you're fetching here `RecordRequest` matching to `Record` pointed by current pk (from url). What if there is no such Request? What if there is more than one such Request? 
IMO `update` on viewset should:
* get `Record` instance
* create `RecordRequest` (without saving) based on data from `Record`
* then perform regular update (or even create) flow (like calling `get_serializer` and `is_valid`)
* instead of calling `perform_update`, we want to call `perform_create` to create new instance of `RecordRequest`

Most of this method would be the same as in `create`.